src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -38,22 +40,24 @@ if (__DEV__) {
 let contextStackCursor : StackCursor<?Object> = createCursor((null: ?Object));
 let didPerformWorkStackCursor : StackCursor<boolean> = createCursor(false);
 
-function getUnmaskedContext() {
-  return contextStackCursor.current || emptyObject;
-}
-
 exports.getMaskedContext = function(workInProgress : Fiber) {
   const type = workInProgress.type;
   const contextTypes = type.contextTypes;
   if (!contextTypes) {
     return emptyObject;
   }
 
-  const unmaskedContext = getUnmaskedContext();
-  const context = {};
+  const hasOwnContext = isContextProvider(workInProgress);
+  // If it is a context provider then use the previous context instead.
```

Comment:
nit: This comment makes sense to me, with the context I have, but might not be sufficient explanation for someone seeing this part of the code for the first time. Maybe we should elaborate about _why_ we use the previous one. Maybe even just a pointer to see the comments in `pushContextProvider`?